language/move-vm/runtime/src/runtime.rs

```diff
@@ -193,6 +201,11 @@ impl VMRuntime {
 
         // All modules verified, publish them to data cache
         for (module, blob) in compiled_modules.into_iter().zip(modules.into_iter()) {
+            if data_store.exists_module(&module.self_id())? {
+                // This is an upgrade, so invalidate the loader cache, which still contains the
+                // old module.
+                self.loader.invalidate();
```

Comment:
Still curious, can we actually just flush the cache here?

Reply:
See top-level comment. Meng did something similar for the epilogue problem if I read this right in https://github.com/diem/diem/pull/7367.